File: apps/mobile/ios/HouseLevi+/Models/AuthToken.swift
import Foundation

struct AuthResponse: Codable {
    let success: Bool; let message: String?; let user: User?
    let token: String?; let accessToken: String?
    let refreshToken: String?; let expiresIn: Int?
    var resolvedAccessToken: String { accessToken ?? token ?? "" }
}
struct AuthToken { let accessToken: String; let refreshToken: String; let expiresIn: Int }
struct LoginRequest: Codable { let email: String; let password: String }
struct RefreshRequest: Codable { let refreshToken: String }
struct OtpRequest: Codable {
    let email: String; let purpose: String
    init(email: String, purpose: String = "login") { self.email = email; self.purpose = purpose }
}
struct OtpVerifyRequest: Codable { let email: String; let otp: String }
struct GenericResponse: Codable { let success: Bool; let message: String?; let error: String? }
struct AuthMeResponse: Codable { let success: Bool; let user: User?; let error: String? }

File: apps/mobile/android/app/src/main/java/com/houselevi/plus/data/models/AuthToken.kt
package com.houselevi.plus.data.models

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    @SerializedName("success")      val success: Boolean     = false,
    @SerializedName("message")      val message: String      = "",
    @SerializedName("user")         val user: User?          = null,
    @SerializedName("token")        val token: String        = "",
    @SerializedName("accessToken")  val accessToken: String  = "",
    @SerializedName("refreshToken") val refreshToken: String = "",
    @SerializedName("expiresIn")    val expiresIn: Long      = 0L,
)
data class AuthToken(val accessToken: String, val refreshToken: String, val expiresIn: Long)
data class LoginRequest(@SerializedName("email") val email: String, @SerializedName("password") val password: String)
data class RefreshRequest(@SerializedName("refreshToken") val refreshToken: String)
data class OtpRequest(@SerializedName("email") val email: String, @SerializedName("purpose") val purpose: String = "login")
data class OtpVerifyRequest(@SerializedName("email") val email: String, @SerializedName("otp") val otp: String)
data class GenericResponse(@SerializedName("success") val success: Boolean = false, @SerializedName("message") val message: String = "")
